lib/redmine_merge_request_links/patches/issue_query_patch.rb

```diff
@@ -0,0 +1,65 @@
+module RedmineMergeRequestLinks
+  module Patches
+    module IssueQueryPatch
+      def self.included(base)
+
+        def initialize_available_filters_with_merge_requests
+          initialize_available_filters_without_merge_requests
+          if User.current.allowed_to?(:view_associated_merge_requests, project, :global => true)
+            add_available_filter("merge_request",
+              :type => :list_optional, :values => ['open', 'merged', 'closed']
+            )
+          end
+        end
+
+        def available_columns_with_merge_requests
+          return @available_columns if @available_columns
+          @available_columns = available_columns_without_merge_requests
+          if User.current.allowed_to?(:view_associated_merge_requests, project, :global => true)
+            @available_columns << QueryColumn.new(:merge_requests)
+          end
+          @available_columns
+        end
+
+        def issues_with_merge_requests(options={})
+          if has_column?(:merge_requests)
+            (options[:include] ||= []) << :merge_requests
```

Comment:
Does this mean merge request are now always eager loaded even when no merge request related columns/filter are active?

Reply:
Ah, the answer is no, right? That's what the `has_column?` call is for.

Reply:
Indeed, only when the column is requested.